staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -476,6 +476,12 @@ func (r *Request) URL() *url.URL {
 	// Join trims trailing slashes, so preserve r.pathPrefix's trailing slash for backwards compatibility if nothing was changed
 	if len(r.resourceName) != 0 || len(r.subpath) != 0 || len(r.subresource) != 0 {
 		p = path.Join(p, r.resourceName, r.subresource, r.subpath)
+		// Since Join trims trailing slashes,
+		// if r.subresource is "proxy" and r.subpath is "/",
+		// we need to preserve the trailing slash to keep the original path.
+		if strings.HasSuffix(r.subpath, "/") {
+			p += "/"
```

Comment:
the comment at line # 476 mentioned "preserve r.pathPrefix's trailing slash", no?

Reply:
I think the meaning of line # 476 is that if `r.resourceName`, `r.subpath` and `r.subresource` are all empty, then do not update variable `p` as `path.Join` trims trailing slashes.